File: Count_Negative_Numbers_in_a_Sorted_Matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int rows = grid.length;
        int cols = grid[0].length;
        int count = 0;

        // Start from the top-right corner of the matrix
        int row = 0;
        int col = cols - 1;

        while (row < rows && col >= 0) {
            // If the current element is negative
            if (grid[row][col] < 0) {
                // Increment the count by the number of negative elements in the current column
                count += rows - row;
                col--;
            } else {
                // Move to the next row
                row++;
            }
        }

        return count;        
    }
}

File: Count_Negative_Numbers_in_a_Sorted_Matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        cols = len(grid[0])
        count = 0
        
        # Start from the top-right corner of the matrix
        row = 0
        col = cols - 1
        
        while row < rows and col >= 0:
            # If the current element is negative
            if grid[row][col] < 0:
                # Increment the count by the number of negative elements in the current column
                count += rows - row
                col -= 1
            else:
                # Move to the next row
                row += 1
        
        return count
